File: base/src/main/java/com/maubis/scarlet/base/database/daos/WidgetDao.java
package com.maubis.scarlet.base.database.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.maubis.scarlet.base.database.entities.Widget;

import java.util.List;

@Dao
public interface WidgetDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  long insert(Widget widget);

  @Delete
  void delete(Widget tag);

  @Query("SELECT * FROM widget WHERE widgetId = :uid LIMIT 1")
  Widget getByID(int uid);

  @Query("SELECT * FROM widget WHERE noteUUID = :uuid")
  List<Widget> getByNote(String uuid);
}

File: base/src/main/java/com/maubis/scarlet/base/database/daos/WidgetDao.kt
package com.maubis.scarlet.base.database.daos

import androidx.room.*
import com.maubis.scarlet.base.database.entities.Widget

@Dao
interface WidgetDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(widget: Widget): Long

    @Delete
    fun delete(tag: Widget)

    @Query("SELECT * FROM widget WHERE widgetId = :uid LIMIT 1")
    fun getByID(uid: Int): Widget?

    @Query("SELECT * FROM widget WHERE noteUuid = :uuid")
    fun getByNote(uuid: String): List<Widget>
}